daemon/oci_linux.go

```diff
@@ -1126,6 +1135,8 @@ func (daemon *Daemon) createSpec(ctx context.Context, daemonCfg *configStore, c
 	}
 	if daemonCfg.Rootless {
 		opts = append(opts, withRootless(daemon, &daemonCfg.Config))
+	} else if userns.RunningInUserNS() {
+		opts = append(opts, withRootfulInRootless(daemon, &daemonCfg.Config))
```

Comment:
quite orthogonal to this PR (sorry for continue asking things - I should do some further digging myself); so the situation with "rootful" in "rootless" is that it doesn't _allow_ setting a score lower than the environment it's running in; but I'm wondering now if it's _desirable_ to allow this in the "non-rootless" situation, i.e. currently this is allowed;

Run a docker-in-docker container with a `500` oom-score adjust;

```bash
docker run -it --rm --privileged -v /var/lib/docker --oom-score-adj 500 docker:dind sh
/# dockerd --debug
```

Within the container, run a container with a oom-score adjust of `100` 

```bash
cat /proc/self/oom_score_adj
500

docker run --rm --oom-score-adj 100 busybox cat /proc/self/oom_score_adj
100
```

In this case (technically) the outer container would be killable before the busybox container (of course, the busybox container would be gone together with it, so it's more of a "theoretical" example), just wondering if we should ultimately go the length, and produce a sensible error for this situation.

Reply:
> but I'm wondering now if it's desirable to allow this in the "non-rootless" situation, i.e. currently this is allowed;

I guess a warning should be printed, but producing an error may break somebody's environment